scipy/_lib/_array_api.py

```diff
@@ -571,3 +587,62 @@ def xp_vector_norm(x: Array, /, *,
     else:
         # to maintain backwards compatibility
         return np.linalg.norm(x, ord=ord, axis=axis, keepdims=keepdims)
+
+
+# work around xp.real raising for floats etc
+def xp_real(a, xp=None):
+    xp = array_namespace(a) if xp is None else xp
+    if xp.isdtype(a.dtype, 'complex floating'):
+        return xp.real(a)
+    else:
+        return a
+
+
+# work around xp.real raising for floats etc
+def xp_imag(a, xp=None):
+    xp = array_namespace(a) if xp is None else xp
+    if xp.isdtype(a.dtype, 'complex floating'):
+        return xp.imag(a)
+    else:
+        return xp.zeros_like(a)
+
+
+# work around xp.real raising for floats etc
+def xp_conj(a, xp=None):
```

Comment:
This one I can imagine could be an annoyance. However the one place this was used it didn't seem needed.